### PR TITLE
Warn on "admin" password

### PR DESCRIPTION
**Pull request by xmo-odoo**
* add support for flash messages on sessions
* show flash messages on the community client's home screen (discuss)

Section 1 of task 31122.

### CODE REVIEW

**Comment by xmo-odoo** [comment]
> looking good, but this means that the flashes will appear on the discuss client action and on the home menu screen (in enterprise). I suppose that it is ok from a functional point of view.

Yeah, given they are the "home pages" of the community & enterprise clients respectively that seemed like a good place (especially given both already have warning messages in various contexts).

**Comment by xmo-odoo** [comment]
@ged-odoo extracted the logic into a proper widget and used messages triggered on the session to allow for & synchronise multiple instances running concurrently, as well as avoid flashes reappearing after they've been dismissed (e.g. open Discuss, note message, dismiss it, go do something in Sales, go back to Discuss, flash message would be back)

File: addons/web/views/webclient_templates.xml [hunk 1]
```diff
@@ -90,6 +90,7 @@
         <script type="text/javascript" src="/web/static/src/js/core/local_storage.js"></script>
         <script type="text/javascript" src="/web/static/src/js/services/local_storage_service.js"></script>
         <script type="text/javascript" src="/web/static/src/js/widgets/rainbow_man.js"></script>
+        <script type="text/javascript" src="/web/static/src/js/widgets/flash.js"></script>
```
Comment (ged-odoo) [inline addons/web/views/webclient_templates.xml]: i think this should go in assets_backend

File: addons/web/static/src/js/widgets/flash.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        session.on('flash_remove', this, this.renderElement);
+        return this._super();
+    },
+    /**
+     * On closing a flash message, notify the session that we need it to
+     * be removed.
+     *
+     * @private
+     */
+    _onClose: function (e) {
```
Comment (ged-odoo) [inline addons/web/static/src/js/widgets/flash.js]: should document that e is a MouseEvent

File: addons/mail/static/src/js/discuss.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ var core = require('web.core');
 var data = require('web.data');
 var Dialog = require('web.Dialog');
 var dom = require('web.dom');
+var FlashMessages = require('web.FlashMessages');
```
Comment (tde-banana-odoo) [inline addons/mail/static/src/js/discuss.js]: Flash is deprecated !

File: addons/web/controllers/main.py [hunk 4]
```diff
@@ -484,6 +484,8 @@ def web_login(self, redirect=None, **kw):
             old_uid = request.uid
             uid = request.session.authenticate(request.session.db, request.params['login'], request.params['password'])
             if uid is not False:
+                if uid == odoo.SUPERUSER_ID and request.params['password'] == 'admin':
```
Comment (tde-banana-odoo) [inline addons/web/controllers/main.py]: Admin is authorized then ?

File: addons/web/static/src/js/widgets/flash.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+odoo.define('web.FlashMessages', function (require) {
+"use strict";
+
+var session = require('web.session');
+var Widget = require('web.Widget');
+
+session.on('flash_remove', session, function (e) {
+    this.flashes.splice(e.data.index, 1);
```
Comment (tde-banana-odoo) [inline addons/web/static/src/js/widgets/flash.js]: Splice girls !

File: odoo/http.py [hunk 6]
```diff
@@ -267,6 +267,10 @@ def session(self):
         """
         return self.httprequest.session
 
+    @lazy_property
+    def flashes(self):
```
Comment (tde-banana-odoo) [inline odoo/http.py]: What do you want to flash ? mmmh ? Are you a flasher ?

**Comment by xmo-odoo** [comment]
After long bitrot and thinking about it more, this might be doable (and better) as a chatter message from the system / superuser / odoobot, adding a flash mechanism seems redundant then.

**Pull request closed without merging**